ui/src/components/view/SearchFilter.vue

```diff
@@ -0,0 +1,619 @@
+            for (const key in json.listhypervisorsresponse.hypervisor) {
+              const hypervisor = json.listhypervisorsresponse.hypervisor[key]
+              if (hypervisor.name === value) {
+                resolve(hypervisor.name)
+              }
+            }
+          }
+        }).catch(() => {
+          resolve(null)
+        })
+      })
+    },
+    getImageStore (storeId) {
+      return new Promise((resolve) => {
+        api('listImageStores', { listAll: true, showicon: true, id: storeId }).then(json => {
+          if (json?.listimagestoresresponse?.imagestore) {
+            resolve(json.listimagestoresresponse.imagestore[0].name)
+          }
+        }).catch(() => {
+          resolve(null)
+        })
+      })
+    },
+    getStoragePool (poolId) {
+      return new Promise((resolve) => {
+        api('listStoragePools', { listAll: true, showicon: true, id: poolId }).then(json => {
+          if (json?.liststoragepoolsresponse?.storagepool) {
+            resolve(json.liststoragepoolsresponse.storagepool[0].name)
+          }
+        }).catch(() => {
+          resolve(null)
+        })
+      })
+    },
+    getPod (podId) {
+      return new Promise((resolve) => {
+        api('listPods', { id: podId }).then(json => {
+          if (json?.listpodsresponse?.pod) {
+            resolve(json.listpodsresponse.pod[0].name)
+          }
+        }).catch(() => {
+          resolve(null)
+        })
+      })
+    },
+    getCluster (clusterId) {
+      return new Promise((resolve) => {
+        api('listClusters', { id: clusterId }).then(json => {
+          if (json?.listclustersresponse?.cluster) {
+            resolve(json.listclustersresponse.cluster[0].name)
+          }
+        }).catch(() => {
+          resolve(null)
+        })
+      })
+    },
+    getInstanceGroup (groupId) {
+      return new Promise((resolve) => {
+        api('listInstanceGroups', { listAll: true, id: groupId }).then(json => {
+          if (json?.listinstancegroupsresponse?.instancegroup) {
+            resolve(json.listinstancegroupsresponse.instancegroup[0].name)
+          }
+        }).catch(() => {
+          resolve(null)
+        })
+      })
+    },
+    getServiceOffering (offeringId) {
+      return new Promise((resolve) => {
+        api('listServiceOfferings', { listAll: true, id: offeringId }).then(json => {
+          if (json?.listserviceofferingsresponse?.serviceoffering) {
+            resolve(json.listserviceofferingsresponse.serviceoffering[0].name)
+          }
+        }).catch(() => {
+          resolve(null)
+        })
+      })
+    },
+    getDiskOffering (offeringId) {
+      return new Promise((resolve) => {
+        api('listDiskOfferings', { listAll: true, id: offeringId }).then(json => {
+          if (json?.listdiskofferingsresponse?.diskoffering) {
+            resolve(json.listdiskofferingsresponse.diskoffering[0].name)
+          }
+        }).catch(() => {
```

Comment:
can have some generic method using resource type for all this listing methods? (maybe, keep/map resource & list api details, and use api for that resource type - will be easy to add in case any other resources are added later)